routes/auth/webAuthn.ts

```diff
@@ -1,75 +1,182 @@
-import { utils } from "ethers";
+import {
+	generateRegistrationOptions,
+	GenerateRegistrationOptionsOpts,
+} from "@simplewebauthn/server";
 import { Request } from "express";
 import { Response } from "express-serve-static-core";
 import { ParsedQs } from "qs";
-import { mintPKP } from "../../lit";
 import {
 	AuthMethodType,
-	AuthMethodVerifyToMintResponse,
-	WebAuthnAssertionVerifyToMintRequest,
+	AuthMethodVerifyRegistrationResponse,
+	WebAuthnVerifyRegistrationRequest,
 } from "../../models";
-import { verifySignature } from "../../utils/webAuthn/verifySignature";
-import { decodeECKeyAndGetPublicKey } from "../../utils/webAuthn/keys";
-import { toUtf8Bytes } from "ethers/lib/utils";
 
-export async function webAuthnAssertionVerifyToMintHandler(
+import type {
+	VerifiedRegistrationResponse,
+	VerifyRegistrationResponseOpts,
+} from "@simplewebauthn/server";
+import { verifyRegistrationResponse } from "@simplewebauthn/server";
+
+import { ethers, utils } from "ethers";
+import { hexlify, toUtf8Bytes } from "ethers/lib/utils";
+import config from "../../config";
+import { getPubkeyForAuthMethod, mintPKP } from "../../lit";
+import { getDomainFromUrl } from "../../utils/string";
+
+/**
+ * Generates WebAuthn registration options for a given username.
+ */
+export function webAuthnGenerateRegistrationOptionsHandler(
+	req: Request<{}, {}, {}, ParsedQs, Record<string, any>>,
+	res: Response<{}, Record<string, any>, number>,
+) {
+	// Get username from query string
+	const username = req.query.username as string | undefined;
+
+	// Get RP_ID from request Origin.
+	const rpID = getDomainFromUrl(req.get("Origin") || "localhost");
+
+	const opts: GenerateRegistrationOptionsOpts = {
+		rpName: "Lit Protocol",
+		rpID,
+		userID: generateUsernameForOptions(username),
```

Comment:
The [spec](https://w3c.github.io/webauthn/#user-handle) recommends that the `userId` be random bytes. Might be worth revisiting in this or the next PR.

Reply:
You mean in the case of usernameless, we use random bytes instead of `Usernameless....` ?

Reply:
Maybe we do `!!username ? username : defaultUsername()` for `userName` field and we do `userID` as the hash of the `userName` field?